tests/sentry/search/events/builder/test_metrics.py

```diff
@@ -2064,6 +2064,48 @@ def test_on_demand_builder_with_new_spec(self):
             # This proves that we're picking up the new spec version
             assert spec_in_use.spec_version.flags == {"include_environment_tag"}
 
+    def test_on_demand_builder_with_not_event_type_error(self):
+        field = "count()"
+        query = "!event.type:error"
```

Comment:
The difference between this test and the next one is that the query is negated and the builder would raise an exception in the next test.